hswidget/fonttablewidget/state.go

```diff
@@ -0,0 +1,80 @@
+package fonttablewidget
+
+import (
+	"fmt"
+
+	"github.com/ianling/giu"
+)
+
+type fontTableWidgetMode int
+
+const (
+	// viewer state
+	fontTableWidgetViewer fontTableWidgetMode = iota
+	// edit rune
+	fontTableWidgetEditRune
+	// opens a dialog and adds a new item on first free index
+	fontTableWidgetAddItem
+)
+
+type widgetState struct {
+	mode     fontTableWidgetMode
+	editRune editRune
+	addItem  addItem
+}
+
+// Dispose cleans state
+func (s *widgetState) Dispose() {
+	s.editRune.Dispose()
+	s.addItem.Dispose()
+}
+
+type editRune struct {
+	editedRune int32
+	startRune  rune
+}
+
+// Dispose disposes a rune state
+func (e *editRune) Dispose() {
+	e.editedRune = rune(0)
+	e.startRune = rune(0)
+}
+
+type addItem struct {
+	newRune editRune
+	width,
+	height int32
+}
```

Comment:
Same goes for this; it sounds like an action instead of a thing.

Reply:
same here: 7d29fa5